### ISSUE TITLE
Compatibility with wordpress.com

### ISSUE THREAD

**Issue by Screenfeed** [wp-media/imagify-plugin#314]
Since our filesystem does tests against the site's root path, Imagify may fail on websites with exotic folder structures, like on wordpress.com, where the ocre is an alias, totally separated from the website's "content".
For example:
- ABSPATH: `/wordpress/core/4.9.8/`,
- Content folder: `/wpcom-123456789/wp-content/`.
In that case, the site's root path should be `/wpcom-123456789/`, but `/wordpress/core/4.9.8/` is returned, so all tests fail.

The good news is that in such environment the folder structure can't be changed. Here is a solution that worked for a support case:
```php
add_filter( 'imagify_site_root', 'imgf_wpcom_site_root', 10001 );
/**
 * Filter the path to the site's root.
 *
 * @since  1.0
 * @author Grégory Viguier
 *
 * @param  string|null $root_path Path to the site's root. Default is null.
 * @return string|null
 */
function imgf_wpcom_site_root( $root_path ) {
	if ( ! defined( 'WPCOMSH_VERSION' ) ) {
		return $root_path;
	}

	$upload_basedir = imagify_get_filesystem()->get_upload_basedir( true );

	if ( strpos( $upload_basedir, '/wp-content/' ) === false ) {
		return $root_path;
	}

	$upload_basedir = explode( '/wp-content/', $upload_basedir );
	$upload_basedir = reset( $upload_basedir );

	return trailingslashit( $upload_basedir );
}
```

[Source](https://secure.helpscout.net/conversation/691521471/83638/).

**Comment by danfitch** [comment]
This is also an issue on Flywheel's Cloud Platform.  The root_path is symlinked into the sites path and causes the creation of the /backups/ folder to fail.

**Comment by Screenfeed** [comment]
@danfitch Thanks for your message.
I'm currently working on something that should handle all cases. When I find some time, I could pack a zip with the fix if you want, so you could give it a try.

**Comment by danfitch** [comment]
@Screenfeed That would be great.  I am a platform engineer and we find quite a few plugins that have similar issues.  I would be glad to forward the zip along and get someone to test it out!

**Comment by Screenfeed** [comment]
@danfitch Fantastic. Here is a [preview of the functionnality](https://www.dropbox.com/s/9nhhptpoznslvv1/imagify-1.8.5-aliased-sites-preview.zip?dl=1).
As for this plugin, it failed so far because we do extra tests on the paths: mainly for security reasons, to prevent messing with files that are outside of the site.
Thanks for your tests 🙂

**Comment by danfitch** [comment]
@Screenfeed I am still getting errors
`The backup folder /www/wp-content/uploads/backup/ cannot be created. Is its parent directory writable by the server? Original images cannot be saved!`
If you have a Flywheel account I can add you as a collaborator to a site.  Or if you have a PR for this code I could take a look at the logic.

**Comment by Screenfeed** [comment]
We have a Flywheel account: wprocket.

The code is not on Github yet, but I can push it if needed. Or if you want to see what I did, take a look at `inc/classes/class-imagify-filesystem.php` from the zip file and compare to the [master](https://raw.githubusercontent.com/wp-media/imagify-plugin/master/inc/classes/class-imagify-filesystem.php) branch, all of the changes are there.

I just tried on the site we use for our tests and it works :|
But from what I see, I think it should even work without these changes.

Side note: probably not related since Imagify doesn't use it, but I see a double slash in `$_SERVER['SCRIPT_FILENAME']`: `/www//wp-admin/admin.php`.
